### PR TITLE
workflows: enable sarif output for CIFuzz

### PR DESCRIPTION
**Pull request by DavidKorczynski**
This will make it easier to interpret the output of CIFuzz by making it possible to view the result in the Github security page.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [N/A] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/pr-fuzz.yaml [hunk 1]
```diff
@@ -24,9 +24,17 @@ jobs:
...
       if: failure() && steps.build.outcome == 'success'
       with:
         name: artifacts
         path: ./out/artifacts
+    - name: Upload Sarif
+      if: always() && steps.build.outcome == 'success'
+      uses: github/codeql-action/upload-sarif@v2
+      with:
+        # Path to SARIF file relative to the root of the repository
+        sarif_file: cifuzz-sarif/results.sarif
```
Comment (patrick-stephens) [inline .github/workflows/pr-fuzz.yaml:39]: I think we should add `category: fuzzing` or similar here?

Comment (DavidKorczynski) [reply .github/workflows/pr-fuzz.yaml:39]: @jonathanmetzman I was thinking about the same thing when going through the docs: https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/uploading-a-sarif-file-to-github#example-workflow-for-sarif-files-generated-outside-of-a-repository -- what's your take on this in the context of CIFuzz?

Comment (patrick-stephens) [reply .github/workflows/pr-fuzz.yaml:39]: I think it's primarily to add more context to the specific result (there may be lots), I did this for the regular scans of the container for example: https://github.com/fluent/fluent-bit/blob/b266d46e5851651f34889a1430491764ed62a074/.github/workflows/cron-trivy.yaml#L78

That way you know where a result comes from when we have potentially lots of results from disparate tools, e.g. code scanning, fuzzing, container CVEs, etc.

Comment (DavidKorczynski) [reply .github/workflows/pr-fuzz.yaml:39]: Sounds good, thanks @patrick-stephens -- I added it!

**Review by patrick-stephens — APPROVED** [review]

**Comment by nokute78** [comment]
This PR may cause CI Fuzz error.
https://github.com/fluent/fluent-bit/actions/runs/5485095155/jobs/9993466873

**Comment by jonathanmetzman** [comment]
Very weird, this has worked in the past. I have a fix here: https://github.com/google/oss-fuzz/pull/10673

**Comment by nokute78** [comment]
Hmm, CI Fuzz detected some errors before merging this patch and they were ignored.
e.g. https://github.com/fluent/fluent-bit/actions/runs/5422155812/jobs/9858454501?pr=7627#step:5:91

This patch may have made it detectable.

**Comment by DavidKorczynski** [comment]
> Hmm, CI Fuzz detected some errors before merging this patch and they were ignored. e.g. https://github.com/fluent/fluent-bit/actions/runs/5422155812/jobs/9858454501?pr=7627#step:5:91
> 
> This patch may have made it detectable.

I don't see an issue in that run? I think the last time CIFuzz detected and issue is here https://github.com/fluent/fluent-bit/pull/7613 and this was due to a bug (fix mentioned in https://github.com/fluent/fluent-bit/pull/7613#issuecomment-1607607534 )

**Comment by jonathanmetzman** [comment]
https://github.com/google/oss-fuzz/pull/10673

**Comment by nokute78** [comment]
@jonathanmetzman @DavidKorczynski Thank you for comment.

Recent CI Fuzz didn't report an error. It is fixed.
e.g. https://github.com/fluent/fluent-bit/actions/runs/5549017264/jobs/10132615553

**Pull request merged**